core/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -168,7 +174,9 @@
         this.queryResult = new QuerySearchResult(id, shardTarget);
         this.fetchResult = new FetchSearchResult(id, shardTarget);
         this.indexShard = indexShard;
+        this.searchExecutor = indexShard.getThreadPool().executor(ThreadPool.Names.SEARCH);
```

Comment:
can we do this on the consumer end and specialize the code in SearchService? I think this is where it's used. We shouldn't really add methods to this class it's big enough